### DISCUSSION TITLE
freecodecamp.org: ads

### DISCUSSION THREAD

**Discussion by RXmanse** [uBlockOrigin/uAssets#23571] [General]
### Prerequisites

- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367), [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) or [_a shortener/hosting site_](https://github.com/uBlockOrigin/uAssets/discussions/17361) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Use this [button](https://user-images.githubusercontent.com/585534/146582579-c32707a0-36a1-4cc5-ad50-83172c9f67a8.png) to find out.)
- [X] I did not remove any of the [default filter lists](https://github.com/uBlockOrigin/uAssets/assets/66517106/6e8d024a-5869-47cf-b142-b2357eb6c7d0), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin along with other content blockers.
- [X] I have verified that the web browser's built-in content blocker/tracking protection, [network wide/DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking), or my VPN is not causing the issue.
- [X] I have verified that other extensions are not causing the issue.
- [X] If this is about a breakage or detection, I have verified that it is caused by uBlock Origin and isn't a site issue.
- [ ] I did not answer truthfully to **ALL** the above checkboxes.

### URL address of the web page

`https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-introductory-javascript-by-building-a-pyramid-generator/step-19`

### Category

ads

### Description

A new type of ad seems to have been introduced to freecodecamp which ridiculously hinders my learning pace, this wasn't there before- atleast it wasn't there the last time I had used the website which was very long time ago.

### Other extensions used

None

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

![image](https://github.com/uBlockOrigin/uAssets/assets/96240924/47b0391a-c225-4632-a68e-237913427f0f)

</details>

### Configuration

<details>

```yaml
uBlock Origin: 1.57.0
Chromium: 121
filterset (summary):
 network: 135036
 cosmetic: 46240
 scriptlet: 20533
 html: 0
listset (total-discarded, last-updated):
 default:
  user-filters: 4-0, never
  easylist: 86331-559, now
  easyprivacy: 50796-811, now
  plowe-0: 3737-0, now
  ublock-badware: 8426-0, 4m
  ublock-filters: 37632-283, 4m
  ublock-privacy: 832-0, 4m
  ublock-quick-fixes: 251-0, 4m
  ublock-unbreak: 2262-0, 4m
  urlhaus-1: 13340-0, now
filterset (user): [array of 4 redacted]
trustedset:
 added: [array of 7 redacted]
userSettings: [none]
hiddenSettings: [none]
supportStats:
 allReadyAfter: 1612 ms (selfie)
 maxAssetCacheWait: 80 ms
 cacheBackend: indexedDB
popupPanel:
 blocked: 3
 network:
  freecodecamp.org: 1
  cloudflareinsights.com: 1
  googletagmanager.com: 1
 extended:
  ##.undefined.text-start.p-\[15px\]
  ##.overflow-scroll.md\:pb-\[30px\].md\:pt-\[30px\].p-\[10px\].ju…
  ##.opacity-50.bg-gray-900.inset-0.fixed
```
</details>

**Comment by ghost** [comment]
I can't reproduce the pop-up. Could you explain how and when did it appear?

**Comment by mapx-** [comment]
@ProcrastinatingDunce 
- do we need an account to reproduce the issue ?
- provide all the steps (original / start link, what to click etc) to get the issue

**Comment by RXmanse** [comment]
> @ProcrastinatingDunce
> @mapx- 
> @voidlyvidevoid 
> * do we need an account to reproduce the issue ?
> * provide all the steps (original / start link, what to click etc) to get the issue

Hey there, I just checked and it seems the ad popup only shows up when you use the website with an account.

In the JavaScript Course, I went up to step 20 without an account and it showed no ads at all whereas when I used an account, I got an ad popup right on step 4.

To replicate this issue, you're going to need to make an account. Then, proceed to any course, I chose JavaScript, after that complete a few steps (I did 4 as you already know), then it should show you an ad, if it doesn't let me know. And I apologize for the late reply, I've been a bit busy with work.

**Comment by ghost** [comment]
`##.h-screen.w-screen.z-1050.relative` did remove it but its self promotion of the website itsself won't be mostly removed (like in https://github.com/uBlockOrigin/uAssets/issues/23403 )

**Comment by mapx-** [comment]
@ProcrastinatingDunce enable uBlock filters – Annoyances and test again

**Comment by RXmanse** [comment]
@mapx- 
I just tried it and it didn't work, the same thing happened, it showed me an ad at step 4.
@voidlyvidevoid 
The self promotion's way too distracting for someone like me, I hope it gets removed but it might not be so.

edit:
@voidlyvidevoid
I added that to my filter lists and tested it, it just broke the website. Like I can see everything but the ad but can't click on anything as if its a blank paper or something.

**Comment by ghost** [comment]
Sorry I can't fix it using any cosmetic filter (as mentioned,it breaks the page). 
<details>
<summary>Screenshot</summary>


![image](https://github.com/uBlockOrigin/uAssets/assets/153749553/c483564b-d827-471b-a23e-f01acd3f393e)


</details>
